testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/sso/FormAuthSSOBypassTestCase.java

```diff
@@ -0,0 +1,103 @@
+package org.jboss.as.test.integration.web.sso;
+
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.utils.HttpClientUtils;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.jboss.as.test.http.util.TestHttpClientUtils;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.test.integration.web.formauth.LogoutServlet;
+import org.jboss.as.test.integration.web.formauth.SecureServlet;
+import org.jboss.as.test.integration.web.formauth.FormAuthUnitTestCase;
+import org.jboss.as.test.integration.web.formauth.SecuredPostServlet;
+import org.jboss.as.test.integration.web.sharedsession.SharedSessionTestCase;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.net.URL;
+
+/**
+ * Tests that if a user is already authenticated via SSO, they are not
+ * challenged with a login form when accessing another application
+ * protected by FORM auth.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(SingleSignOnSetup.class)
+public class FormAuthSSOBypassTestCase {
+
+    @ArquillianResource
+    private URL baseURL;
+
+    @Deployment(name = "sso-form-bypass.ear", testable = false)
+    public static EnterpriseArchive deployment() {
+
+        JavaArchive appClasses = ShrinkWrap.create(JavaArchive.class, "auth-classes.jar")
```

Comment:
There is no need for new arbitrary Strings. 

```suggestion
        JavaArchive appClasses = ShrinkWrap.create(JavaArchive.class, FormAuthSSOBypassTestCase.class.getSimpleName() + ".jar")
```